ECIPs/ECIP-1027.md

```diff
@@ -0,0 +1,218 @@
+same Geth instance.
+
+## Key points:
+
+ - There should be a special Smart Contract deployed to Parent Chain to manage checkpoints
+ - Checkpoints are transactions in Parent Chain
+ - Sidechain can read from Parent Chain and verify that it follows longest or most difficult Sidechain
+ - Mainnet is going to work as usual. Not a fork.
+
+It maybe helpful to introduce additional Opcodes to improve cross-transaction functionality. That will require a 
+protocol upgrade, but that’s optional and will be discussed in separate ECIPs
+
+## Types of chains
+
+There can be selected major 3 types of Sidechains:
+
+ - _Private chain_ - uses mainnet only for consensus
+ - _Trusted public chain_ - operated by a trusted party
+ - _Public chain_ - general use blockchain operated in untrusted environment
+
+### Private Chain (C1)
+
+Private Chain is a fully private chain executed in isolated environment which is using Parent Chain only as a security 
+mechanism.
+ 
+This can be a solution for many cases of a private corporate blockchains. This types of chain doesn’t share state with 
+parent chain, and therefore doesn’t need cross transactions, or can perform them in individual way when needed.
+
+Conception of making checkpoints on public secure blockchain is a popular idea in private chains and standalone 
+services, also called as Anchoring.
+
+### Trusted Public Chain (C2)
+
+Trusted Public Chain is a chain managed by a public service or Dapp, and such service provides full management of 
+that chain and performs cross operations for their users. Sidechain can use PoA or PoS for internal consensus, and a 
+separate Oracle/Validator like service to send/confirm user actions performed across the chains. For most of the cases 
+this chain could have a simple shared state, such as buying/selling internal tokens through a parent coin.
+ 
+Smart contracts for controlling shared state, and Transaction Replay like logic can give users additional security for 
+their cross-chain operations.
+
+_Example_: a smart contract based Token Exchange with Deposit/Withdraw through a special Smart Contract in Parent chain
+
+### Public Chain (C3)
+
+Public Chain is a general use blockchain, operated by multiple parties and in an untrusted environment. It can use a 
+fast PoS consensus algorithm for fast and cheap transactions, or individual configuration of EVM.
+
+## Checkpoint authorization
+
+Making a checkpoint will require verification mechanism to avoid corrupted state and to protect from malicious actors.
+
+For simple cases, when Sidechain is solely managed by one business (C1/C2 types), it’s enough to authorize an address 
+(or set of addresses) which can make such checkpoints.
+
+For more sophisticated chains (C2/C3 types) verification a Sidechain can use methods similar to Proof Of Stake, or 
+even implement full verification in a smart contract. Later is possible if a checkpoint can provide security by 
+validating only a chain headers and part of shared state.
```

Comment:
I find that section most interesting - really how to connect two blockchain together and to reach consensus, but you left technical details beyond the scope of the document. What about the next steps to work out some working solution for this?